src/scripts/docker_migration_apt.sh

```diff
@@ -0,0 +1,168 @@
+#!/bin/bash
+
+# Set environment variables to avoid interactive prompts
+export DEBIAN_FRONTEND=noninteractive
+DOCKER_DOWNLOAD_ORIGINS="Docker:\${distro_codename}"
+UNATTENDED_UPGRADES_FILE="/etc/apt/apt.conf.d/50unattended-upgrades"
+LOG_FILE="/usr/src/dappnode/logs/upgrade_015.log"
+
+log() {
+  echo "$(date '+%Y-%m-%d %H:%M:%S') $1" | tee -a ${LOG_FILE}
+}
+
+log "Starting docker install migration from pkg to apt"
+
+# Only update docker if unattended upgrades is enabled,
+# otherwise the update might crash due to updating docker from docker.
+if [ ! -f "${UNATTENDED_UPGRADES_FILE}" ]; then
+  log "WARNING: Unattended upgrades is not enabled, skipping upgrade"
+  exit 0
+fi
+
+# Check if docker is installed via apt
+# The docker.list file is created by the docker installation script
+if [ -f /etc/apt/sources.list.d/docker.list ]; then
+  log "Docker is already installed via apt, skipping upgrade"
+  # TODO: check if docker is in unattended upgrades
+  exit 0
```

Comment:
Be careful here, with docker installed via apt you will not add it to the unattended upgrades if you exit here